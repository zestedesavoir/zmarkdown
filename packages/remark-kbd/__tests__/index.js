import dedent from 'dedent'
import {unified} from 'unified'
import reParse from 'remark-parse'
import kbdPlugin from '../lib/index'
import remarkStringify from 'remark-stringify'
import rehypeStringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'
// TODO (next): reintroduce once the plugin is ready
//import remarkCustomBlocks from '../../remark-custom-blocks'

const render = text => unified()
  .use(reParse)
  /*.use(remarkCustomBlocks, {
    secret: 'spoiler',
  })*/
  .use(kbdPlugin)
  .use(remark2rehype)
  .use(rehypeStringify)
  .processSync(text)

const fixture = dedent`
  Blabla ||ok|| kxcvj ||ok foo|| sdff

  sdf |||| df

  sfdgs | | dfg || dgsg | qs

  With two pipes: \||key|| you'll get ||key||.

  It cannot contain inline markdown:

  * ||hell[~~o~~](#he)?||

  It cannot contain blocks:

  * ||hello: [[secret]]?||
`

describe('parses kbd', () => {
  it('parses a big fixture', () => {
    const {value} = render(fixture)
    expect(value).toMatchSnapshot()
  })

  it('escapes the start marker', () => {
    const {value} = render(dedent`
      ||one|| \||escaped|| ||three|| \|||four|| ||five||
    `)
    expect(value).toContain('||escaped||')
    expect(value).toContain('|<kbd>four</kbd>')
  })
})

test('allow non-pipe characters', () => {
  const {value} = unified()
    .use(reParse)
    .use(kbdPlugin, {char: '+'})
    .use(remark2rehype)
    .use(rehypeStringify)
    .processSync('++CTRL++, \\+++D++')

  expect(value).toMatchSnapshot()
})

test('to markdown', () => {
  const {value} = unified()
    .use(reParse)
    .use(remarkStringify)
    .use(kbdPlugin)
    .processSync(fixture)

  expect(value).toMatchSnapshot()
})
