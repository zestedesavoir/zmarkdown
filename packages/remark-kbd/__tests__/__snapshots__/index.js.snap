// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allow non-pipe characters 1`] = `"<p><kbd>CTRL</kbd>, +<kbd>D</kbd></p>"`;

exports[`parses kbd parses a big fixture 1`] = `
"<p>Blabla <kbd>ok</kbd> kxcvj <kbd>ok foo</kbd> sdff</p>
<p>sdf |||| df</p>
<p>sfdgs | | dfg || dgsg | qs</p>
<p>With two pipes: ||key|| you'll get <kbd>key</kbd>.</p>
<p>It cannot contain inline markdown:</p>
<ul>
<li><kbd>hell[~~o~~](#he)?</kbd></li>
</ul>
<p>It cannot contain blocks:</p>
<ul>
<li><kbd>hello: [[secret]]?</kbd></li>
</ul>"
`;

exports[`to markdown 1`] = `
"Blabla ||ok|| kxcvj ||ok foo|| sdff

sdf \\|\\|\\|\\| df

sfdgs \\| \\| dfg \\|\\| dgsg \\| qs

With two pipes: \\|\\|key\\|\\| you'll get ||key||.

It cannot contain inline markdown:

* ||hell\\[~~o~~]\\(#he)?||

It cannot contain blocks:

* ||hello: \\[\\[secret]]?||
"
`;
